### PR TITLE
impl_ele_funcs_floor_ceil_trunc

### PR DESCRIPTION
**Pull request by vtavana**
In this PR, the following elementwise functions are implemented: `dptctl.tensor.floor`, `dpctl.tensor.ceil`, and `dpctl.tensor.trunc`.

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to an issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [X] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: dpctl/tensor/libtensor/include/kernels/elementwise_functions/ceil.hpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    resT operator()(const argT &in)
+    {
+        if constexpr (std::is_integral_v<argT>) {
+            return in;
+        }
+        else {
+            return sycl::ceil(in);
+        }
+        // return sycl::ceil(in);
```
[the remark this answers is not in the document]
Comment (ndgrigorian) [reply dpctl/tensor/libtensor/include/kernels/elementwise_functions/ceil.hpp:74]: Never mind, I was mistaken, this case is not required, and doesn't make sense to begin with.

File: dpctl/tensor/libtensor/include/kernels/elementwise_functions/ceil.hpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+template <typename T> struct CeilOutputType
+{
+    using value_type = typename std::disjunction< // disjunction is C++17
+                                                  // feature, supported by DPC++
+        td_ns::TypeMapResultEntry<T, bool, sycl::half>,
+        td_ns::TypeMapResultEntry<T, sycl::half>,
+        td_ns::TypeMapResultEntry<T, float>,
+        td_ns::TypeMapResultEntry<T, double>,
+        td_ns::DefaultResultEntry<void>>::result_type;
```
[the remark this answers is not in the document]
Comment (ndgrigorian) [reply dpctl/tensor/libtensor/include/kernels/elementwise_functions/ceil.hpp:97]: Boolean type not required by array API. Returned array must have same type for all other types, including integer.

Please apply this change to ``floor`` and ``trunc`` and adjust tests accordingly.

**Comment by oleksandr-pavlyk** [comment]
The output type for of integer type inputs should be the appropriate floating point type.

```
Python 3.9.12 (main, Jun  1 2022, 11:38:51)
Type 'copyright', 'credits' or 'license' for more information
IPython 8.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numpy as np

In [2]: np.floor.types
Out[2]: ['e->e', 'f->f', 'd->d', 'f->f', 'd->d', 'g->g', 'O->O']

In [3]: np.ceil.types
Out[3]: ['e->e', 'f->f', 'd->d', 'f->f', 'd->d', 'g->g', 'O->O']

In [4]: np.trunc.types
Out[4]: ['e->e', 'f->f', 'd->d', 'f->f', 'd->d', 'g->g', 'O->O']

In [5]: import dpctl.tensor as dpt

In [6]: import dpctl.tensor._tensor_impl as ti

In [7]: ti._ceil_result_type(dpt.int64)
Out[7]: dtype('int64')

In [8]: dpt.ceil(dpt.ones(10, dtype="uint32"))
Out[8]: usm_ndarray([1, 1, 1, 1, 1, 1, 1, 1, 1, 1], dtype=uint32)

In [9]: np.ceil(np.ones(10, dtype="uint32"))
Out[9]: array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
```

**Comment by vtavana** [comment]
> The output type for of integer type inputs should be the appropriate floating point type.
> 
> ```
> Python 3.9.12 (main, Jun  1 2022, 11:38:51)
> Type 'copyright', 'credits' or 'license' for more information
> IPython 8.4.0 -- An enhanced Interactive Python. Type '?' for help.
> 
> In [1]: import numpy as np
> 
> In [2]: np.floor.types
> Out[2]: ['e->e', 'f->f', 'd->d', 'f->f', 'd->d', 'g->g', 'O->O']
> 
> In [3]: np.ceil.types
> Out[3]: ['e->e', 'f->f', 'd->d', 'f->f', 'd->d', 'g->g', 'O->O']
> 
> In [4]: np.trunc.types
> Out[4]: ['e->e', 'f->f', 'd->d', 'f->f', 'd->d', 'g->g', 'O->O']
> 
> In [5]: import dpctl.tensor as dpt
> 
> In [6]: import dpctl.tensor._tensor_impl as ti
> 
> In [7]: ti._ceil_result_type(dpt.int64)
> Out[7]: dtype('int64')
> 
> In [8]: dpt.ceil(dpt.ones(10, dtype="uint32"))
> Out[8]: usm_ndarray([1, 1, 1, 1, 1, 1, 1, 1, 1, 1], dtype=uint32)
> 
> In [9]: np.ceil(np.ones(10, dtype="uint32"))
> Out[9]: array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
> ```

Yes, the behavior is different than `numpy` since Array API for [ceil](https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.ceil.html), [floor ](https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.floor.html)and [trunc ](https://data-apis.org/array-api/2022.12/API_specification/generated/array_api.trunc.html)says that "the returned array must have the same data type as input."

**Comment by oleksandr-pavlyk** [comment]
Yes, I asked for clarification: https://github.com/data-apis/array-api/issues/657

**Comment by oleksandr-pavlyk** [comment]
Why are these tests so slow?

```
(dev_dpctl) opavlyk@opavlyk-mobl:~/repos/dpctl$ python -m pytest --durations=6 dpctl/tests/elementwise/test_floor_ceil_trunc.py
============================================================================ test session starts ============================================================================
platform linux -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/opavlyk/repos/dpctl
plugins: metadata-2.0.4, json-report-1.5.0, cov-4.0.0, hypothesis-6.75.3, scikit-learn-1.3.dev0
collected 186 items

dpctl/tests/elementwise/test_floor_ceil_trunc.py ..............................sss.......................................sss.................................sss..... [ 62%]
.........................sss..............................sss......sss                                                                                                [100%]

============================================================================ slowest 6 durations ============================================================================
7.31s call     dpctl/tests/elementwise/test_floor_ceil_trunc.py::test_floor_ceil_trunc_usm_type[host-trunc-UnaryElementwiseFunc]
6.96s call     dpctl/tests/elementwise/test_floor_ceil_trunc.py::test_floor_ceil_trunc_usm_type[shared-ceil-UnaryElementwiseFunc]
6.49s call     dpctl/tests/elementwise/test_floor_ceil_trunc.py::test_floor_ceil_trunc_usm_type[host-floor-UnaryElementwiseFunc]
6.46s call     dpctl/tests/elementwise/test_floor_ceil_trunc.py::test_floor_ceil_trunc_usm_type[shared-trunc-UnaryElementwiseFunc]
6.34s call     dpctl/tests/elementwise/test_floor_ceil_trunc.py::test_floor_ceil_trunc_usm_type[host-ceil-UnaryElementwiseFunc]
5.97s call     dpctl/tests/elementwise/test_floor_ceil_trunc.py::test_floor_ceil_trunc_usm_type[shared-floor-UnaryElementwiseFunc]
================================================================ 168 passed, 18 skipped in 103.48s (0:01:43) ================================================================
```

I am going to push changes to these tests to make them run faster.

**Pull request merged**